spirv_parser.cpp

```diff
@@ -771,6 +771,8 @@ void Parser::parse(const Instruction &instruction)
 
 		uint32_t result_type = ops[0];
 		uint32_t id = ops[1];
+		auto &type = get<SPIRType>(result_type);
```

Comment:
Would make more sense to me if the code that queries load_type_width would check for `maybe_get<SPIRVariable>` and `maybe_get<SPIRConstant>` before falling back to normal SSA variable query.

Reply:
If I understand you correctly, what you're asking for is to only track for IDs and get the types at the function `get_case_list()`. I tried following this approach but usually when the `block.condition` points to an `OpLoad`, the `type_id` ends up being a `TypeNone` which makes visually impossible to get the actual type (with helpers like `expression_type` etc.). That's why I've decided to track the type that early on. If you see any other way please let me know!

Reply:
Yes, but certain things are defined up front, like PHI nodes (SPIRVariable), constants and spec constants (SPIRConstant). SPIRExpression is found late as you say, and we'll need the map for that scenario.